packages/ra-ui-materialui/src/input/ReferenceInput.tsx

```diff
@@ -84,14 +84,11 @@ export const ReferenceInput = (props: ReferenceInputProps) => {
         filter,
     });
 
-    if (props.validate) {
+    if (props.validate && process.env.NODE_ENV !== 'production') {
         throw new Error(
             '<ReferenceInput> does not accept a validate prop. Set the validate prop on the child instead.'
         );
     }
-    if (Children.count(children) !== 1) {
-        throw new Error('<ReferenceInput> only accepts a single child');
-    }
```

Comment:
**Question:** Why did you remove this test?

Also, this creates a linter warning at line 1.

Reply:
I removed it because it was only required when we cloned the child. We no longer do that, so there is no need to force a single child.

Reply:
Also, I fixed the warning